gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -683,7 +683,8 @@ public static Builder builder() {
   BucketInfo update(BucketInfo bucketInfo, BucketTargetOption... options);
 
   /**
-   * Update blob information.
+   * Update blob information. All the blob's metadata will be replaced using values from the given
+   * {@code blobInfo}. Update will fail if an incomplete {@code blobInfo} is provided.
```

Comment:
> Do you understand why create is not failing if a user supply content-type (as the apiary api for insert suggests this is a required value)?

I don't. I tried explicitly supplying a `null` contentType but still does not fail. Seems like a default value is used but I cannot understand where that comes from.

> For update I assumed we are using an empty acl, list. I find it strange to force providing the ACL list when the "insert" does not force it. Did you consider a default content-type here for consistency with create?

`BlobInfo` default Acl is `null` not an empty list. In addition to that, providing an empty list still causes and "Invalid Argument" error. Providing a non-empty ACL list solves the problem. Ideas?

> Also, for all update operations can we make it even more clear that "this will replace all the blob information and if that is not desired, a patch should be used instead"?

Sure thing!